src/nvim/option.c

```diff
@@ -4368,35 +4368,8 @@ int makeset(FILE *fd, int opt_flags, int local_only)
             cmd = "setlocal";
           }
 
-          if (option_has_type(opt_idx, kOptValTypeBoolean)) {
-            if (put_setbool(fd, cmd, opt->fullname, *(int *)varp) == FAIL) {
-              return FAIL;
-            }
-          } else if (option_has_type(opt_idx, kOptValTypeNumber)) {
-            if (put_setnum(fd, cmd, opt->fullname, (OptInt *)varp) == FAIL) {
-              return FAIL;
-            }
-          } else {    // string
-            bool do_endif = false;
-
-            // Don't set 'syntax' and 'filetype' again if the value is
-            // already right, avoids reloading the syntax file.
-            if (opt->indir == PV_SYN || opt->indir == PV_FT) {
-              if (fprintf(fd, "if &%s != '%s'", opt->fullname,
-                          *(char **)(varp)) < 0
```

Comment:
Where did this check go?

Reply:
@echasnovski How necessary is this check? Removing it removes a fair amount of complexity, and tests seem tp pass just fine with it. Seems like it's just for performance? If so, unless keeping that check is _really_ worth the added complexity, I'm in favor of removing it.

Reply:
I'll add it back for now